src/SignalR/server/Core/src/Internal/DefaultHubDispatcher.cs

```diff
@@ -749,4 +776,38 @@ public override IReadOnlyList<Type> GetParameterTypes(string methodName)
 
         return null;
     }
+
+    private static Activity? CreateActivity(IServiceProvider serviceProvider, string methodName)
+    {
+        if (serviceProvider.GetService<ActivitySource>() is ActivitySource activitySource
+            && activitySource.HasListeners())
+        {
+            var requestContext = Activity.Current?.Context;
+            // Get off the parent span.
+            // This is likely the Http Request span and we want Hub method invocations to not be collected under a long running span.
+            Activity.Current = null;
```

Comment:
What happens when the activity created by this method is complete? Is the originalrequest activity restored? I think it should be.

Reply:
It already should be restored, that's how async locals work. And the link check in the test(s) verifies this.

Reply:
This feels like the wrong place to do this.

Reply:
We could do it once in `HubConnectionHandler.OnConnectedAsync` and store the original activity in a field so we can access it for the linking.

Reply:
Do we want the linking?

Reply:
Since we're removing our activities from the parent span, the whole websocket request, it'd be nice to point to the parent via a link so there is some correlation. Also, the default OnConnectedAsync span we were talking about below becomes less useful if you can't see when the request started.

Reply:
That makes sense.

Reply:
> It already should be restored, that's how async locals work. And the link check in the test(s) verifies this.

I would expect `Activity.Stop()` to set `Activity.Current` to its parent when called, but if we clear out the parent before creating this activity and then call stop, how does it know to restore the old parent?

Reply:
> how does it know to restore the old parent?

It wouldn't. I'm guessing @BrennanConroy was referring to the implicit restoration of async locals that occurs when an async method returns to its caller? That does work but it feels like a subtle dependency that could easily be broken in some future refactoring. Ideally I'd clear Activity.Current somewhere up the stack to make it obvious what scope of code is executing dissociated from the long-running HTTP server activity.

Reply:
Changed where we clear the parent as mentioned in https://github.com/dotnet/aspnetcore/pull/55439#discussion_r1588473527

This still relies on async local behavior to reset the parent if the user runs code after the SignalR middleware runs, but that seems fine? Unless we want to explicitly reset it when `HubConnectionHandler.OnConnectedAsync` finishes.